+nosnoc/+model/Objects.m

```diff
@@ -0,0 +1,421 @@
+% BSD 2-Clause License
+
+% Copyright (c) 2023, Armin Nurkanović, Jonathan Frey, Anton Pozharskiy, Moritz Diehl
+
+% Redistribution and use in source and binary forms, with or without
+% modification, are permitted provided that the following conditions are met:
+
+% 1. Redistributions of source code must retain the above copyright notice, this
+%    list of conditions and the following disclaimer.
+
+% 2. Redistributions in binary form must reproduce the above copyright notice,
+%    this list of conditions and the following disclaimer in the documentation
+%    and/or other materials provided with the distribution.
+
+% THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+% AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+% IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+% DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+% FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+% DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+% SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+% CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+% OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+% OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+% This file is part of NOSNOC.
+classdef Objects < nosnoc.model.Base
+    properties (Access=public)
+        
+        % Algebraics
+        lambda
+        mu
+        alpha
+        p_d
+        y1_d
+        y2_d
+        lambda_t
+        v_t
+        normal_lift
+        gamma_f
+
+        % x_dot lift
+        x_dot_lift
+
+        % friction expressions
+        G_friction
+        H_friction
+        g_friction
+
+        % Feasible set
+        r = 0.01
+        g_d
+        g_kkt
+        normal
+        normal0
+        tangent
+        tangent0
+
+        % Distance functions
+        c
+
+        % objects
+        objects
+        contacts
+
+    end
+
+    methods (Access=public)
+        function obj = Objects()
+            dims = struct;
+        end
+
+        function verify_and_backfill(obj, opts)
+            import casadi.*
+            verify_and_backfill@nosnoc.model.Base(obj,opts);
+
+            dims = obj.dims;
+            
+            dims.n_c = size(obj.c, 1);
+            dims.n_lambda = dims.n_c;
+            dims.n_y1d = size(obj.y1_d, 1);
```

Comment:
?
described what is what, otherwise no one knows how to interpret the variables and their features